api/lib/infrastructure/repositories/campaign-participation-repository.js

```diff
@@ -52,28 +59,33 @@ module.exports = {
     return BookshelfCampaignParticipation
       .where({ userId })
       .orderBy('createdAt', 'DESC')
-      .fetchAll({ withRelated: ['campaign'] })
+      .fetchAll({ withRelated: ['campaign', 'assessment'] })
       .then((bookshelfCampaignParticipation) => bookshelfCampaignParticipation.models)
       .then(fp.map(_toDomain));
   },
 
   findOneByAssessmentIdWithSkillIds(assessmentId) {
     return BookshelfCampaignParticipation
-      .where({ assessmentId })
       .query((qb) => {
         qb.innerJoin('campaigns', 'campaign-participations.campaignId', 'campaigns.id');
         qb.innerJoin('target-profiles', 'campaigns.targetProfileId', 'target-profiles.id');
         qb.innerJoin('target-profiles_skills', 'target-profiles.id', 'target-profiles_skills.targetProfileId');
+        qb.innerJoin('assessments', 'assessments.campaignParticipationId', 'campaign-participations.id');
+        qb.where('assessments.id', '=', assessmentId);
       })
-      .fetch({ required: false, withRelated: ['campaign.targetProfile.skillIds'] })
+      .fetch({ required: false, withRelated: ['campaign.targetProfile.skillIds', 'assessment'] })
       .then(_convertToDomainWithSkills);
   },
 
   findByAssessmentId(assessmentId) {
     return BookshelfCampaignParticipation
-      .where({ assessmentId })
-      .fetchAll({ withRelated: ['campaign', 'user'] })
-      .then((campaignParticipations) => bookshelfToDomainConverter.buildDomainObjects(BookshelfCampaignParticipation, campaignParticipations));
+      .query((qb) => {
+        qb.innerJoin('assessments', 'assessments.campaignParticipationId', 'campaign-participations.id');
```

Comment:
Même remarque que plus haut sur le fait que la jointure peut multiplier les lignes.

Reply:
Cf https://github.com/1024pix/pix/pull/676/commits/236027615639f823039c0dc8e9853c42017dad51#r321392110

Reply:
Du coup je réitère ma solution écrite aussi plus haut, virer la jointure et spécifier le withRelated

Reply:
Même réponse que plus haut du coup ;)